unicode-data/lib/Unicode/Char/Case.hs

```diff
@@ -47,3 +77,153 @@ isUpperCase = P.isUppercase
 {-# DEPRECATED isUpper "Use isUpperCase instead. Note that the behavior of this function does not match base:Data.Char.isUpper. See Unicode.Char.Case.Compat for behavior compatible with base:Data.Char." #-}
 isUpper :: Char -> Bool
 isUpper = P.isUppercase
+
+-- $case
+--
+-- Correct case conversion rules may map one input character to two or three
+-- output characters. For examples, see the documentation of 'toCaseFoldString',
+-- 'toLowerString', 'toTitleString' and 'toUpperString'.
+--
+-- __Note:__ In some languages, case conversion is a locale- and
+-- context-dependent operation. The case conversion functions in this
+-- module are /not/ locale nor context sensitive.
+
+-- [TODO] @since
+-- | Returns the full /folded/ case mapping of a character.
+--
+-- It uses the character property @Case_Folding@.
+--
+-- __Note:__ @\'\\NUL\'@ requires special handling.
+-- See 'toCaseFoldString' source code for an example.
+caseFoldingMapping :: Unfold Char Char
```

Comment:
Add INLINE.

How about renaming to `caseFoldMapping` to keep it consistent with other names like `lowerCaseMapping`?

Reply:
Done